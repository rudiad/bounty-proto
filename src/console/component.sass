.console
  bottom: 0
  position: absolute
  border: 1px solid black
  padding: .7em
  left: 0
  right: 0
  margin-left: auto
  margin-right: auto
  height: 10em
  color: white
  background-color: #241d30
  opacity: 0.8
  overflow-y: auto
  font-family: "Lucida Console", Monaco, monospace
  z-index: -1
