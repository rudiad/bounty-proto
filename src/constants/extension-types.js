const { ExtensionAnchor, ExtensionPlatform } = window['extension-coordinator'];

const extensionTypes = {};
extensionTypes[ExtensionAnchor.Overlay] = "Overlay";
extensionTypes[ExtensionAnchor.Panel] = "Panel";
extensionTypes[ExtensionAnchor.Component] = "Component";
extensionTypes[ExtensionPlatform.Mobile] = "Mobile";

export const ExtensionAnchors = Object.freeze(extensionTypes);
export const DEFAULT_EXTENSION_TYPE = ExtensionAnchor.Overlay;
