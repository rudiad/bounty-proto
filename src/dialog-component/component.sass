$top-bar-dimension: 40px

.dialog__backdrop
  position: fixed
  top: 0
  bottom: 0
  left: 0
  right: 0
  background-color: #19171C
  opacity: 0.75
  padding: 50
  height: 100vh

.dialog
  background-color: white
  border-radius: 5
  width: 66%
  margin: 0 auto
  border: 1px solid #DAD8DE
  position: absolute
  top: 20%
  left: 20%

.container
  display: flex

.container__block
  display: inline-block

.container__left-block
  @extend .container__block
  float: left
  margin-left: 20px

.container__right-block
  @extend .container__block
  float: right

.top-bar
  @extend .container
  height: $top-bar-dimension
  justify-content: space-between

.top-bar-element
  width: auto
  text-align: center
  vertical-align: middle
  line-height: $top-bar-dimension
  font-weight: lighter

.top-bar-element-left
  @extend .container__left-block
  @extend .top-bar-element

.top-bar-element-right
  @extend .container__right-block
  @extend .top-bar-element

.dialog__divider-width
  width: 96%

@mixin bottom-bar-box($color, $bg-color)
  color: $color
  background-color: $bg-color
  border: 1px solid #7D55C7
  float: left
  text-align: center
  width: 80px
  height: 30px
  vertical-align: middle
  line-height: 30px
  cursor: pointer
  font-size: 16px
  margin-left: 20px
  margin-top: 10px
  margin-bottom: 30px

.title
  font-size: 13px
  font-weight: lighter
