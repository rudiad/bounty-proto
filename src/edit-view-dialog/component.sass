@import "../dialog-component/component.sass"

.edit-view__background
  @extend .dialog__backdrop

.edit-view__dialog-component
  @extend .dialog
  height: 275px
  width: 275px
  left: 40%

.edit-view__dialog-mobile
  @extend .dialog
  height: 275px
  width: 415px
  left: 35%

.edit-view__dialog-overlay
  @extend .dialog
  height: 275px
  width: 275px
  left: 40%

.edit-view__content
  @extend .container__left-block
  overflow-y: auto
  height: 50%
  width: 95%
  display: flex
  flex-direction: column

.edit-subcontainer__input
  margin-top: .5em
  margin-bottom: .5em

.dialog__top-bar-container
  @extend .top-bar

.top-bar-container__title
  @extend .top-bar-element-left
  margin-top: 5px

.top-bar-container__escape
  @extend .top-bar-element-right
  width: $top-bar-dimension
  cursor: pointer
  width: 4em
  margin-top: 5px

.dialog__divider
  @extend .dialog__divider-width

.dialog__bottom-bar
  @extend .container__left-block

.bottom-bar__save
  @include bottom-bar-box(white, #7D55C7)

.bottom-bar__cancel
  @include bottom-bar-box(#7D55C7, white)
  margin-left: 10px
