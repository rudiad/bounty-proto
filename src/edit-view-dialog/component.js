import React, { Component } from 'react';
import PropTypes from 'prop-types';
import closeButton from '../img/close_icon.png';
import './component.sass';

export class EditViewDialog extends Component {
  constructor(args) {
    super(args);
    this.state = {
      x: 0,
      y: 0,
    }
  }

  onChange = (input) => {
    const newState = {};
    newState[input.target.name] = input.target.value;
    this.setState(newState);
  }

  componentDidMount() {
    this.props.views.forEach(element => {
      if (element.id === this.props.idToEdit) {
        this.setState({
          x: element.x,
          y: element.y,
        });
      }
    });
  }

  render() {
    if (!this.props.show) {
      return null;
    }

    return (
      <div className="edit-view">
        <div className="edit-view__background"/>
        <div className="edit-view__dialog">
          <div className="dialog__top-bar-container">
            <div className="top-bar-container__title"> Edit View </div>
            <div className="top-bar-container__escape" onClick={this.props.closeHandler}><img alt="Close" src={closeButton}/></div>
          </div>
          <hr className="dialog__divider"/>
          <div className="edit-view__content">
            <div className="type-and-size-container__type-title">
                  Position (%)
            </div>
              <div className='overlay-custom-container'>
                <div className='overlay-custom-container'>
                <div className="edit-subcontainer__input">
                  <label className="inputs__option-label">X</label>
                  <input type="text" name="x" value={this.state.x} onChange={this.onChange}/>
                </div>
                <div className="edit-subcontainer__input">
                  <label className="inputs__option-label">Y</label>
                  <input type="text" name="y" value={this.state.y} onChange={this.onChange}/>
                </div>
                </div>
              </div>
            </div>
          <hr className="dialog__divider"/>
          <div className="dialog_bottom-bar">
            <div className="bottom-bar__save" onClick={() => this.props.saveViewHandler({ x: this.state.x, y: this.state.y })}> Save </div>
            <div className="bottom-bar__cancel" onClick={this.props.closeHandler}> Cancel </div>
          </div>
        </div>
      </div>
    );
  }
}

EditViewDialog.propTypes = {
  show: PropTypes.bool.isRequired,
  idToEdit: PropTypes.string.isRequired,
  views: PropTypes.array.isRequired,
  closeHandler: PropTypes.func.isRequired,
  saveViewHandler: PropTypes.func.isRequired,
};
