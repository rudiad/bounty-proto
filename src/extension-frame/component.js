import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

const { ExtensionPlatform, ExtensionViewType} = window['extension-coordinator'];

const IFRAME_CLASS = 'extension-frame';
const EXTENSION_FRAME_INIT_ACTION = 'extension-frame-init';

export class ExtensionFrame extends Component {
  componentDidMount() {
    if (this.iframe) {
      this.iframe.onload = this._extensionFrameInit;
    }
  }

  render() {
    return (
      <iframe
      ref={this._bindIframeRef}
        src={process.env.PUBLIC_URL + '/extension-frame.html'}
        frameBorder={0}
        className={'rig-frame ' + IFRAME_CLASS}
        title={this.props.frameId}/>
    );
  }

  _bindIframeRef = (iframe) => {
    this.iframe = iframe;
  }

  _extensionFrameInit = () => {
    const extension = {
      anchor: this.props.type,
      channelId: this.props.extension.channelId,
      loginId: null,
      extension: this.props.extension,
      mode: this.props.mode,
      platform: (this.props.type === ExtensionViewType.Mobile) ? ExtensionPlatform.Mobile : ExtensionPlatform.Web,
      trackingProperties: {},
      iframeClassName: IFRAME_CLASS,
    }
    const data = {
      extension: extension,
      action: EXTENSION_FRAME_INIT_ACTION,
      frameId: this.props.frameId,
    }
    this.iframe.contentWindow.postMessage(data, '*');
  }

  _onIdentityLinked(isLinked) {}

  _onFrameDoubleClick(evt) {
    evt.preventDefault();
  }

  _onModalRequested(confirmationRequest) {}
}

ExtensionFrame.propTypes = {
  className: PropTypes.string.isRequired,
  frameId: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
};
