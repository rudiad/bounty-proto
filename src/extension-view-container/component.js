import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ExtensionView } from '../extension-view';
import { ExtensionViewButton } from '../extension-view-button';
import { EXTENSION_MODE_TO_VIEW } from '../constants/extension-modes.js'
import './component.sass';
const { ExtensionMode } = window['extension-coordinator'];

export class ExtensionViewContainer extends Component {
  openExtensionViewDialog = () => {
    this.props.openExtensionViewHandler();
  }

  render() {
    if (this.props.mode !== ExtensionMode.Viewer) {
      return ( <ExtensionView
          id={this.props.mode}
          type={EXTENSION_MODE_TO_VIEW[this.props.mode]}
          extension={this.props.extension}
          mode={this.props.mode}
          key={this.props.mode}/>
      );
    }

    let extensionViews = [];
    if (this.props.extensionViews) {
      extensionViews = this.props.extensionViews.map(view => {
        return <ExtensionView
          key={view.id}
          id={view.id}
          extension={view.extension}
          type={view.type}
          mode={this.props.mode}
          role={view.role}
          overlaySize={view.overlaySize}
          position={{x: view.x, y: view.y}}
          linked={view.linked}
          openEditViewHandler={this.props.openEditViewHandler}
          deleteViewHandler={this.props.deleteExtensionViewHandler}/>
      });
    }

    return (
      <div className='view-container-wrapper'>
          <div className="view-container">
          { extensionViews }
        </div>
        <div>
          <ExtensionViewButton
            onClick={this.openExtensionViewDialog}>
          </ExtensionViewButton>
        </div>
      </div>
    );
  }
}

ExtensionViewContainer.propTypes = {
  mode: PropTypes.string.isRequired,
  extensionViews: PropTypes.array.isRequired,
  openEditViewHandler: PropTypes.func,
  deleteExtensionViewHandler: PropTypes.func.isRequired,
  openExtensionViewHandler: PropTypes.func.isRequired,
  extension: PropTypes.object.isRequired,
}
