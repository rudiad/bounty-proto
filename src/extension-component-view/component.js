import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ExtensionFrame } from '../extension-frame';
const { ExtensionMode, ExtensionViewType, getComponentPositionFromView, getComponentSizeFromView } = window['extension-coordinator'];

export class ExtensionComponentView extends Component {
  computeViewStyles() {
    const extension = this.props.extension;
    const positionFromView = getComponentPositionFromView(
      this.props.overlaySize.width,
      this.props.overlaySize.height,
      {
        x: this.props.position.x * 100,
        y: this.props.position.y * 100,
      });
    const sizeFromView = getComponentSizeFromView(
      this.props.overlaySize.width,
      this.props.overlaySize.height,
      extension.views.component);

    let viewStyles = {
      border: '1px solid #7D55C7',
      position: 'absolute',
      left: positionFromView.x + 'px',
      top: positionFromView.y + 'px',
      width: `${sizeFromView.width}px`,
      height: `${sizeFromView.height}px`,
    }

    if (extension.views.component.zoom) {
      viewStyles = {
        width: `${sizeFromView.width / sizeFromView.zoomScale}px`,
        height: `${sizeFromView.height / sizeFromView.zoomScale}px`,
        transformOrigin: '0 0',
        transform: `scale(${sizeFromView.zoomScale})`,
      }
    }

    return viewStyles;
  }

  render() {
    return (
      <div
        className="view component-view"
        style={{
          width: this.props.overlaySize.width + 'px',
          height: this.props.overlaySize.height + 'px',
        }}>
          <div style={this.computeViewStyles()}>
          <ExtensionFrame
            className="view"
            frameId={`frameid-${this.props.id}`}
            extension={this.props.extension}
            type={ExtensionViewType.Component}
            mode={ExtensionMode.Viewer}
          />
        </div>
      </div>
    );
  }
}

ExtensionComponentView.propTypes = {
  id: PropTypes.string.isRequired,
  extension: PropTypes.object.isRequired,
  overlaySize: PropTypes.object.isRequired,
  position: PropTypes.object.isRequired,
  role: PropTypes.string,
};
