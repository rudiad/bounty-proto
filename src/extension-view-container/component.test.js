import { setupShallowTest } from '../tests/enzyme-util/shallow';
import { ExtensionForTest, createViewsForTest } from '../tests/constants/extension';
import { ExtensionViewContainer } from './component';
import { ExtensionAnchors } from '../constants/extension-types';
import { ViewerTypes } from '../constants/viewer-types';
const { ExtensionMode, ExtensionAnchor } = window['extension-coordinator'];

describe('<ExtensionViewContainer />', () => {
  const setupShallow = setupShallowTest(ExtensionViewContainer, () => ({
    mode: ExtensionMode.Viewer,
    extensionViews: createViewsForTest(0, '', ''),
    deleteExtensionViewHandler: jest.fn(),
    openExtensionViewHandler: jest.fn(),
    openEditViewHandler: jest.fn(),
    extension: ExtensionForTest
  }));

  it('openExtensionViewHandler is called when the create button is clicked', () => {
    const { wrapper } = setupShallow();
    wrapper.find('ExtensionViewButton').simulate('click');
    expect(wrapper.instance().props.openExtensionViewHandler).toHaveBeenCalled();
   });

  describe('when in viewer mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow();
      expect(wrapper).toMatchSnapshot();
    });
    it('has the correct number of views', () => {
      const { wrapper } = setupShallow({
        extensionViews: createViewsForTest(2, ExtensionAnchors[ExtensionAnchor.Panel], ViewerTypes.LoggedOut)
      });
      expect(wrapper.find('ExtensionView')).toHaveLength(2);
    });
  });

  describe('when in broadcaster config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        mode: ExtensionMode.Config
      });
      expect(wrapper).toMatchSnapshot();
    });
    it('has the correct number of views', () => {
      const { wrapper } = setupShallow({
        mode: ExtensionMode.Config
      });
      expect(wrapper.find('ExtensionView')).toHaveLength(1);
    });
  });

  describe('when in live config mode', () => {
    it('renders correctly', () => {
      const { wrapper } = setupShallow({
        mode: ExtensionMode.Dashboard
      });
      expect(wrapper).toMatchSnapshot();
    });
    it('has the correct number of views', () => {
      const { wrapper } = setupShallow({
        mode: ExtensionMode.Dashboard
      });
      expect(wrapper.find('ExtensionView')).toHaveLength(1);
    });
  });
});
