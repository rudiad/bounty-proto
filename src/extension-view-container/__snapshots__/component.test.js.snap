// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionViewContainer /> when in broadcaster config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionViewContainer
    deleteExtensionViewHandler={[Function]}
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    extensionViews={Array []}
    mode="config"
    openEditViewHandler={[Function]}
    openExtensionViewHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "config",
    "nodeType": "class",
    "props": Object {
      "extension": Object {
        "authorName": "test",
        "channelId": "channelId",
        "description": "description",
        "iconUrl": "icon_url",
        "id": "id",
        "name": "name",
        "requestIdentity": false,
        "sku": "sku",
        "state": "state",
        "summary": "summary",
        "token": "token",
        "vendorCode": "vendorCode",
        "version": "0.1",
        "views": Object {
          "component": Object {
            "aspectHeight": 3000,
            "aspectWidth": 2500,
            "viewerUrl": "test",
            "zoom": false,
          },
          "config": Object {
            "viewerUrl": "test",
          },
          "liveConfig": Object {
            "viewerUrl": "test",
          },
          "panel": Object {
            "viewerUrl": "test",
          },
        },
        "whitelistedConfigUrls": Array [
          "foo",
        ],
        "whitelistedPanelUrls": Array [
          "bar",
        ],
      },
      "id": "config",
      "mode": "config",
      "type": "config",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "config",
      "nodeType": "class",
      "props": Object {
        "extension": Object {
          "authorName": "test",
          "channelId": "channelId",
          "description": "description",
          "iconUrl": "icon_url",
          "id": "id",
          "name": "name",
          "requestIdentity": false,
          "sku": "sku",
          "state": "state",
          "summary": "summary",
          "token": "token",
          "vendorCode": "vendorCode",
          "version": "0.1",
          "views": Object {
            "component": Object {
              "aspectHeight": 3000,
              "aspectWidth": 2500,
              "viewerUrl": "test",
              "zoom": false,
            },
            "config": Object {
              "viewerUrl": "test",
            },
            "liveConfig": Object {
              "viewerUrl": "test",
            },
            "panel": Object {
              "viewerUrl": "test",
            },
          },
          "whitelistedConfigUrls": Array [
            "foo",
          ],
          "whitelistedPanelUrls": Array [
            "bar",
          ],
        },
        "id": "config",
        "mode": "config",
        "type": "config",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionViewContainer /> when in live config mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionViewContainer
    deleteExtensionViewHandler={[Function]}
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    extensionViews={Array []}
    mode="dashboard"
    openEditViewHandler={[Function]}
    openExtensionViewHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "dashboard",
    "nodeType": "class",
    "props": Object {
      "extension": Object {
        "authorName": "test",
        "channelId": "channelId",
        "description": "description",
        "iconUrl": "icon_url",
        "id": "id",
        "name": "name",
        "requestIdentity": false,
        "sku": "sku",
        "state": "state",
        "summary": "summary",
        "token": "token",
        "vendorCode": "vendorCode",
        "version": "0.1",
        "views": Object {
          "component": Object {
            "aspectHeight": 3000,
            "aspectWidth": 2500,
            "viewerUrl": "test",
            "zoom": false,
          },
          "config": Object {
            "viewerUrl": "test",
          },
          "liveConfig": Object {
            "viewerUrl": "test",
          },
          "panel": Object {
            "viewerUrl": "test",
          },
        },
        "whitelistedConfigUrls": Array [
          "foo",
        ],
        "whitelistedPanelUrls": Array [
          "bar",
        ],
      },
      "id": "dashboard",
      "mode": "dashboard",
      "type": "liveConfig",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "dashboard",
      "nodeType": "class",
      "props": Object {
        "extension": Object {
          "authorName": "test",
          "channelId": "channelId",
          "description": "description",
          "iconUrl": "icon_url",
          "id": "id",
          "name": "name",
          "requestIdentity": false,
          "sku": "sku",
          "state": "state",
          "summary": "summary",
          "token": "token",
          "vendorCode": "vendorCode",
          "version": "0.1",
          "views": Object {
            "component": Object {
              "aspectHeight": 3000,
              "aspectWidth": 2500,
              "viewerUrl": "test",
              "zoom": false,
            },
            "config": Object {
              "viewerUrl": "test",
            },
            "liveConfig": Object {
              "viewerUrl": "test",
            },
            "panel": Object {
              "viewerUrl": "test",
            },
          },
          "whitelistedConfigUrls": Array [
            "foo",
          ],
          "whitelistedPanelUrls": Array [
            "bar",
          ],
        },
        "id": "dashboard",
        "mode": "dashboard",
        "type": "liveConfig",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ExtensionViewContainer /> when in viewer mode renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtensionViewContainer
    deleteExtensionViewHandler={[Function]}
    extension={
        Object {
            "authorName": "test",
            "channelId": "channelId",
            "description": "description",
            "iconUrl": "icon_url",
            "id": "id",
            "name": "name",
            "requestIdentity": false,
            "sku": "sku",
            "state": "state",
            "summary": "summary",
            "token": "token",
            "vendorCode": "vendorCode",
            "version": "0.1",
            "views": Object {
              "component": Object {
                "aspectHeight": 3000,
                "aspectWidth": 2500,
                "viewerUrl": "test",
                "zoom": false,
              },
              "config": Object {
                "viewerUrl": "test",
              },
              "liveConfig": Object {
                "viewerUrl": "test",
              },
              "panel": Object {
                "viewerUrl": "test",
              },
            },
            "whitelistedConfigUrls": Array [
              "foo",
            ],
            "whitelistedPanelUrls": Array [
              "bar",
            ],
          }
    }
    extensionViews={Array []}
    mode="viewer"
    openEditViewHandler={[Function]}
    openExtensionViewHandler={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="view-container"
>
          
</div>,
        <div>
          <ExtensionViewButton
                    onClick={[Function]}
          />
</div>,
      ],
      "className": "view-container-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "view-container",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ExtensionViewButton
            onClick={[Function]}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="view-container"
>
            
</div>,
          <div>
            <ExtensionViewButton
                        onClick={[Function]}
            />
</div>,
        ],
        "className": "view-container-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "view-container",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ExtensionViewButton
              onClick={[Function]}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
